src/openqaoa-core/algorithms/baseworkflow.py

```diff
@@ -0,0 +1,578 @@
+#   Copyright 2022 Entropica Labs
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from abc import ABC
+from typing import List
+import json
+import gzip
+from os.path import exists
+
+from .workflow_properties import BackendProperties, ClassicalOptimizer
+from ..backends import DeviceLocal, DeviceBase
+from ..problems import QUBO
+from ..utilities import delete_keys_from_dict, is_valid_uuid, generate_uuid
+from ..backends.qaoa_backend import (
+    DEVICE_NAME_TO_OBJECT_MAPPER,
+    DEVICE_ACCESS_OBJECT_MAPPER,
+)
+
+
+class Workflow(ABC):
```

Comment:
Why do we call it `Workflow`? The folder is called `algorithms`, meaning that `QAOA` and `RQAOA` are `Algorithms` not `Workflows`, why don't we get rid of the word 'workflow'? Didn't we want to do that?

Reply:
Raised issue in #182 continue discussion there.